libraries/tools/kotlin-stdlib-gen/src/templates/Arrays.kt

```diff
@@ -187,5 +185,26 @@ fun arrays(): List<GenericFunction> {
         }
     }
 
+    templates add f("nullIfEmpty()") {
+        only(ArraysOfObjects, ArraysOfPrimitives, Collections)
+        doc { f ->
+            """
+            Returns `this` value if it is not empty or `null`, if it is.
+
+            ${if (f == ArraysOfObjects || f == ArraysOfPrimitives)
+                "@sample samples.collections.Arrays.nullIfEmpty"
+            else if (f == Collections)
+                "@sample samples.collections.Collections.nullIfEmpty"
+            else
+                ""
+            }
+            """
+        }
+        returns("SELF?")
+        body {
+            "return if (this == null || this.isEmpty()) null else this"
```

Comment:
You're right. I mixed it with the `String`/`CharSequence` implementation.
Should I unify both to apply on non-nullable type?
I will also revert the formatting changes.